src/platform/macos_iokit/enumeration.rs

```diff
@@ -36,11 +53,30 @@ fn usb_service_iter() -> Result<IoServiceIterator, Error> {
 }
 
 pub fn list_devices() -> Result<impl Iterator<Item = DeviceInfo>, Error> {
-    Ok(usb_service_iter()?.filter_map(probe_device))
+    Ok(usb_service_iter(kIOUSBDeviceClassName)?.filter_map(probe_device))
+}
+
+pub fn list_buses() -> Result<impl Iterator<Item = BusInfo>, Error> {
+    // Chain all the HCI types into one iterator
+    // A bit of a hack, could maybe probe IOPCIDevice and filter on children with IOClass.starts_with("AppleUSB")
+    Ok(usb_service_iter(kAppleUSBXHCI)?
+        .filter_map(|h| probe_bus(h, UsbController::XHCI))
+        .chain(
+            usb_service_iter(kAppleUSBEHCI)?
+                .filter_map(|h| probe_bus(h, UsbController::EHCI))
+                .chain(
+                    usb_service_iter(kAppleUSBOHCI)?
+                        .filter_map(|h| probe_bus(h, UsbController::OHCI))
+                        .chain(
+                            usb_service_iter(kAppleUSBVHCI)?
+                                .filter_map(|h| probe_bus(h, UsbController::VHCI)),
+                        ),
+                ),
+        ))
```

Comment:
You mean like?

```
    Ok([
        usb_service_iter(kAppleUSBXHCI)?
            .filter_map(|h| probe_bus(h, UsbControllerType::XHCI))
            .collect::<Vec<_>>(),
        usb_service_iter(kAppleUSBEHCI)?
            .filter_map(|h| probe_bus(h, UsbControllerType::EHCI))
            .collect::<Vec<_>>(),
        usb_service_iter(kAppleUSBOHCI)?
            .filter_map(|h| probe_bus(h, UsbControllerType::OHCI))
            .collect::<Vec<_>>(),
        usb_service_iter(kAppleUSBVHCI)?
            .filter_map(|h| probe_bus(h, UsbControllerType::VHCI))
            .collect::<Vec<_>>(),
    ]
    .into_iter()
    .flatten())
```

I was under the impression the chain is better because it keeps the iterator lazy? Or maybe I misunderstood and you're not suggesting collecting each step.